flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/reader/RowDataReaderFunction.java

```diff
@@ -39,7 +39,7 @@ public class RowDataReaderFunction extends DataIteratorReaderFunction<RowData> {
   private final EncryptionManager encryption;
 
   public RowDataReaderFunction(
-      Configuration config, Schema tableSchema, Schema projectedSchema,
+      ReadableConfig config, Schema tableSchema, Schema projectedSchema,
```

Comment:
switch to `ReadableConfig` interface. since this is not released, we can still make the breaking change.